internal/messagequeue/messagequeue.go

```diff
@@ -450,24 +451,25 @@ func (mq *MessageQueue) simulateDontHaveWithTimeout(msg bsmsg.BitSwapMessage) {
 	mq.dhTimeoutMgr.AddPending(wants)
 }
 
-// func (mq *MessageQueue) logOutgoingMessage(msg bsmsg.BitSwapMessage) {
-// 	entries := msg.Wantlist()
-// 	for _, e := range entries {
-// 		if e.Cancel {
-// 			if e.WantType == pb.Message_Wantlist_Have {
-// 				log.Debugf("send %s->%s: cancel-have %s\n", lu.P(mq.network.Self()), lu.P(mq.p), lu.C(e.Cid))
-// 			} else {
-// 				log.Debugf("send %s->%s: cancel-block %s\n", lu.P(mq.network.Self()), lu.P(mq.p), lu.C(e.Cid))
-// 			}
-// 		} else {
-// 			if e.WantType == pb.Message_Wantlist_Have {
-// 				log.Debugf("send %s->%s: want-have %s\n", lu.P(mq.network.Self()), lu.P(mq.p), lu.C(e.Cid))
-// 			} else {
-// 				log.Debugf("send %s->%s: want-block %s\n", lu.P(mq.network.Self()), lu.P(mq.p), lu.C(e.Cid))
-// 			}
-// 		}
-// 	}
-// }
+func (mq *MessageQueue) logOutgoingMessage(msg bsmsg.BitSwapMessage) {
+	self := mq.network.Self()
+	entries := msg.Wantlist()
+	for _, e := range entries {
```

Comment:
Instead of writing a bunch of separate log messages, lets:

1. Store a non-suggared logger somewhere (`sflog`): https://godoc.org/go.uber.org/zap#SugaredLogger.Desugar
2. Log the message using a single log entry using `sflog.Check(log.DebugLevel, "sending message")` to avoid doing _anything_ unless debug logging is enabled. We can log arrays of "things" (canceled wants, new wants, etc.).

Reply:
I added a check against the current log level 👍 

I think it will be easier to eyeball if they're separate messages, and we'll probably only output this level of logging when we're tracing exactly what bitswap is doing. So my preference would be to leave them as separate messages.